src/app/functions/server/getBreaches.ts

```diff
@@ -51,46 +51,53 @@ export async function getBreachIcons (breaches: Breach[]): Promise<LogoMap> {
     return new Map()
   }
   isFetchingIcons = true
-  const breachDomains = breaches
-    .map(breach => breach.Domain)
-    .filter(breachDomain => breachDomain.length > 0)
-  const logoFolder = pathResolve(dirname(fileURLToPath(import.meta.url)), '../../../../public/logo_cache/')
-  try {
-    await mkdir(logoFolder)
-  } catch {
-    // Do nothing; if the directory already exists, that's fine.
-  }
-  const existingLogos = await readdir(logoFolder)
-  // TODO: Batch to limit memory use?
-  const logoMapElems = await Promise.all(breachDomains.map(breachDomain => {
-    return new Promise((resolve, reject) => {
-      const logoFilename = breachDomain.toLowerCase() + '.ico'
-      const logoPath = pathResolve(logoFolder, logoFilename)
-      if (existingLogos.includes(logoFilename)) {
-        resolve([breachDomain, `/logo_cache/${logoFilename}`])
-        return
-      }
-      get(`https://icons.duckduckgo.com/ip3/${breachDomain}.ico`, (response) => {
-        if (response.statusCode !== 200) {
-          resolve(null)
+  async function fetchBreachIcons () {
+    const breachDomains = breaches
+      .map(breach => breach.Domain)
+      .filter(breachDomain => breachDomain.length > 0)
```

Comment:
Unrelated, but possibly interesting aside&hellip;
Several (15?) Domains have been breached 2+ times. Not sure if that means it's possible we'll request some site logos multiple times, or if the `async`/`await` Promises would resolve the first request and skip the other requests for the same logo.

```sh
curl --silent https://haveibeenpwned.com/api/v3/breaches | npx prettier --parser=json | grep '"Domain":' | sort | uniq -c | grep -Ev "\s+1\s+" | sort

   2     "Domain": "abandonia.com",
   2     "Domain": "adultfriendfinder.com",
   2     "Domain": "bell.ca",
   2     "Domain": "cardmafia.cc",
   2     "Domain": "cprewritten.net",
   2     "Domain": "data4marketers.com",
   2     "Domain": "flashflashrevolution.com",
   2     "Domain": "forum.btcsec.com",
   2     "Domain": "linkedin.com",
   2     "Domain": "mgmresorts.com",
   2     "Domain": "r2games.com",
   2     "Domain": "twitter.com",
   2     "Domain": "wishbone.io",
   2     "Domain": "zoosk.com",
   4     "Domain": "ogusers.com",
  27     "Domain": "",
```

Again, possibly not interesting, or a blocker, but probably more of a nit/microoptimization.